tests/lisa/test_wlgen.py

```diff
@@ -0,0 +1,379 @@
+        return [args[0] if args else kwargs['command']
+                for args, kwargs in self.execute_calls]
+
+    def clear_execute_calls(self):
+        self.execute_calls = []
+
+class LisaSelfBase(TestCase):
+    """
+    Base class for LISA self-tests
+
+    Creates and sets up a TestTarget.
+
+    Provides directory paths to use for output files. Deletes those paths if
+    they already exist, to try and provide a clean test environment. This
+    doesn't create those paths, tests should create them if necessary.
+    """
+
+    tools = []
+    """Tools to install on the 'target' before each test"""
+
+    @property
+    def target_run_dir(self):
+        """Unique directory to use for creating files on the 'target'"""
+        return os.path.join(self.target.working_directory,
+                            'lisa_target_{}'.format(self.__class__.__name__))
+
+    @property
+    def host_out_dir(self):
+        """Unique directory to use for creating files on the host"""
+        return os.path.join(
+            os.getenv('LISA_HOME'), 'results',
+            'lisa_selftest_out_{}'.format(self.__class__.__name__))
+
+    def setUp(self):
+        self.target = TestTarget()
+
+        tools_path = os.path.join(os.getenv('LISA_HOME'),
+                                  'tools', self.target.abi)
+        self.target.setup([os.path.join(tools_path, tool)
+                           for tool in self.tools])
+
+        if self.target.directory_exists(self.target_run_dir):
+            self.target.remove(self.target_run_dir)
+
+        if os.path.isdir(self.host_out_dir):
+            shutil.rmtree(self.host_out_dir)
+
+        self.target.clear_execute_calls()
+
+class RTABase(LisaSelfBase):
+    """
+    Common functionality for testing RTA
+
+    Doesn't have "Test" in the name so that nosetests doesn't try to run it
+    directly
+    """
+
+    tools = ['rt-app']
+
+    def get_expected_command(self, rta_wload):
+        """Return the rt-app command we should execute when `run` is called"""
+        rta_path = os.path.join(self.target.executables_directory, 'rt-app')
+        json_path = os.path.join(rta_wload.run_dir, rta_wload.json)
+        return '{} {} 2>&1'.format(rta_path, json_path)
+
+    def setUp(self):
+        super(RTABase, self).setUp()
+
+        # Can't calibrate rt-app because:
+        # - Need to set performance governor
+        # - Need to use SCHED_FIFO + high priority
+        # We probably don't have permissions so use a dummy calibration.
+        self.calibration = {c: 100
+                           for c in range(len(self.target.cpuinfo.cpu_names))}
+
+        os.makedirs(self.host_out_dir)
+
+    def assert_output_file_exists(self, path):
+        """Assert that a file was created in host_out_dir"""
+        path = os.path.join(self.host_out_dir, path)
+        self.assertTrue(os.path.isfile(path),
+                        'No output file {} from rt-app'.format(path))
+
+class TestRTAProfile(RTABase):
```

Comment:
I can see this source becoming bigger and bigger in short time... maybe it's worth to setup it since the beginning to be a more scalable solution?

What about moving RTA* and Perf* classes into dedicated classes in the same folder?
Do you think it could be possible?

I would propose:
```
tests/lisa/test_wlgen.py
tests/lisa/test_wlgen_rtapp.py
tests/lisa/test_wlgen_perf.py
```